src/misc/utils.cc

```diff
@@ -279,18 +291,23 @@ int32_t ExecCommandSync(const string& cmdline, vector<string> args,
       break;
     case 0: {
       // Child
-      // Close parent pipe descriptors
-      close(infd[1]);
-      close(outfd[0]);
-      close(errfd[0]);
+      int fd, fds;
+
       // set up pipes
       CHECK(dup2(infd[0], STDIN_FILENO) == STDIN_FILENO);
       CHECK(dup2(outfd[1], STDOUT_FILENO) == STDOUT_FILENO);
       CHECK(dup2(errfd[1], STDERR_FILENO) == STDERR_FILENO);
-      // close unnecessary pipe descriptors
-      close(infd[0]);
-      close(outfd[1]);
-      close(errfd[1]);
+
+      // Close all file descriptors other than stdin, stdout and stderr
```

Comment:
I'm a little confused: why do we do this? There shouldn't usually be any others open, but there could be (e.g. if a perf file is written or logging is done concurrently). Is this in order to get rid of stale open FDs?